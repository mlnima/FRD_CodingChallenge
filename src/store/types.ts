export const GET_VEHICLE_MANUFACTURERS ='GET_VEHICLE_MANUFACTURERS'
export const GET_VEHICLE_MANUFACTURER ='GET_CAR_MANUFACTURER'
export const GET_VEHICLE_MODEL_DATA ='GET_VEHICLE_MODEL_DATA'
export const SET_ALERT ='SET_ALERT'
export const SET_LAST_UPDATE ='SET_LAST_UPDATE'
export const LOG_ERRORS ='LOG_ERRORS'
export const SET_ERROR ='SET_ERROR'
export const REMOVE_ERROR_FROM_STORE ='REMOVE_ERROR_FROM_STORE'
export const CHANGE_VEHICLE_FILTER ='CHANGE_VEHICLE_FILTER'
export const SET_VEHICLE_FILTERS_OPTIONS ='SET_VEHICLE_FILTERS_OPTIONS'
export const SET_VEHICLE_FILTERS_DATA ='SET_VEHICLE_FILTERS_DATA'
export const SET_GOT_RESULT='SET_GOT_RESULT'



