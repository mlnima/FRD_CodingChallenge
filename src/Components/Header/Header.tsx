import React from "react";
import Logo from "./Logo/Logo";
import styled from "styled-components";
import Navigation from "./Navigation/Navigation";

const HeaderStyledHeader = styled.header`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  height: 68px;
  background-color: var( --header-background-color,#ccc);
  width: 100%;
`

const Header :React.FC = () => {
    return (
        <HeaderStyledHeader id={'header'} className={'header'}>
            <Logo/>
            <Navigation/>
        </HeaderStyledHeader>
    )
};
export default Header
