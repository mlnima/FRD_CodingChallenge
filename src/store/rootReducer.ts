import {combineReducers} from "redux";
import {globalStateReducer} from "./reducers/globalReducer";
import {vehicleModelPageReducer} from "./reducers/vehicleModelPageReducer";

export default combineReducers({
    globalState:globalStateReducer,
    vehicleModelState:vehicleModelPageReducer,
})