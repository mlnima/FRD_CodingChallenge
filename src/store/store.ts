import {applyMiddleware, compose, createStore} from "redux";
import thunk from "redux-thunk";
import reducer from './rootReducer';
const middleware = [thunk];

export const store = createStore(reducer, compose(applyMiddleware(...middleware)));

