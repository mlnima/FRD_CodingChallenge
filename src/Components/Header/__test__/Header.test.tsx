import React from 'react';
import {render} from '@testing-library/react';
import {store} from "../../../store/store";
import {Provider} from 'react-redux';
import Header from '../Header';
import {BrowserRouter} from "react-router-dom";

describe('Header', () => {
    test('renders Header component', () => {

        const { container } = render(
            <BrowserRouter>
                <Provider store={store}>
                    <Header />
                </Provider>
            </BrowserRouter>
        );

        const headerElement = container.querySelector('#header')
        expect(headerElement).toBeInTheDocument();
        expect(headerElement).toHaveClass('header');
        expect(headerElement).toMatchSnapshot()

    });

});