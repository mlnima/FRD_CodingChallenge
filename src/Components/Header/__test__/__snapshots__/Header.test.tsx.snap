// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header component 1`] = `
<header
  class="sc-dkPtRN bJuTIo header"
  id="header"
>
  <a
    class="sc-bdvvtL gYXAjx"
    href="/"
    title="friday"
  >
    <img
      alt="logo"
      class="logo-image"
      src="/images/logo.png"
    />
  </a>
  <nav
    class="sc-gsDKAQ CsWXP"
  >
    <a
      class="navigation-item"
      href="/vehicleManufacturers"
    >
      Vehicle Manufacturers
    </a>
  </nav>
</header>
`;
